bugbear.py

```diff
@@ -229,3 +254,13 @@ def compose_call_path(self, node):
             "to the exception.",
     type=BugBearChecker,
 )
+
+B901 = partial(
+    error,
+    message=("B901: Using ``yield`` together with ``return x``. Use ``raise "
+             "StopIteration(x)`` instead of ``return x`` if this was "
+             "intentional."),
```

Comment:
Well, if the user *really* uses `return x` correctly and intentionally in a generator, how is she supposed to let the linter know?

If you add `# noqa` to the line, you are letting both the linter and the human reader know that you are in fact doing this on purpose.